File: src/main/java/io/github/wntopia/gikipedia/server/domain/auth/dto/response/DataGsmStudentInfoResDto.java
package io.github.wntopia.gikipedia.server.domain.auth.dto.response;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import java.io.Serializable;

@JsonIgnoreProperties(ignoreUnknown = true)
public record DataGsmStudentInfoResDto(
    Long id,
    String name,
    String sex,
    String email,
    Integer grade,
    Integer classNum,
    Integer number,
    Integer studentNumber,
    String major,
    String specialty,
    String role,
    Integer dormitoryFloor,
    Integer dormitoryRoom,
    Boolean isLeaveSchool,
    DataGsmClubSummaryResDto majorClub,
    DataGsmClubSummaryResDto autonomousClub,
    String githubId,
    String githubUrl)
    implements Serializable {}

File: src/main/kotlin/io/github/wntopia/gikipedia/server/domain/auth/dto/response/DataGsmStudentInfoResDto.kt
package io.github.wntopia.gikipedia.server.domain.auth.dto.response

import com.fasterxml.jackson.annotation.JsonIgnoreProperties
import java.io.Serializable

@JsonIgnoreProperties(ignoreUnknown = true)
data class DataGsmStudentInfoResDto(
    val id: Long?,
    val name: String?,
    val sex: String?,
    val email: String?,
    val grade: Int?,
    val classNum: Int?,
    val number: Int?,
    val studentNumber: Int?,
    val major: String?,
    val specialty: String?,
    val role: String?,
    val dormitoryFloor: Int?,
    val dormitoryRoom: Int?,
    val isLeaveSchool: Boolean?,
    val majorClub: DataGsmClubSummaryResDto?,
    val autonomousClub: DataGsmClubSummaryResDto?,
    val githubId: String?,
    val githubUrl: String?,
) : Serializable
